# `complete-sentences-line-comments`

Requires complete sentences for multi-line leading line comments.

## Rule Details

```ts
// Bad

// Good
```

## Options and Defaults

```json
{
  "rules": {
    "complete/complete-sentences-line-comments": "error"
  }
}
```

This rule is not configurable.

## Resources

- [How to use this rule](../..)
- [Rule source](https://github.com/complete-ts/complete/blob/main/packages/eslint-plugin-complete/src/rules/complete-sentences-line-comments.ts)
- [Test source](https://github.com/complete-ts/complete/blob/main/packages/eslint-plugin-complete/tests/rules/complete-sentences-line-comments.test.ts)
