# complete/require-ascii

ğŸ’¼ This rule is enabled in the âœ… `recommended` config.

ğŸ“ Require code and comments to only contain ASCII characters.

ğŸ’­ This rule requires [type information](https://typescript-eslint.io/linting/typed-linting).

<!-- end auto-generated rule header -->

Require code and comments to only contain ASCII characters.

## Rule Details

```ts
// Bad

// Good
```

## Options

This rule is not configurable.

## Resources

- [How to use this rule](https://complete-ts.github.io/eslint-plugin-complete)
- [Rule source](https://github.com/complete-ts/complete/blob/main/packages/eslint-plugin-complete/src/rules/require-ascii.ts)
- [Test source](https://github.com/complete-ts/complete/blob/main/packages/eslint-plugin-complete/tests/rules/require-ascii.test.ts)
